File: market-application/src/main/java/com/example/market/application/port/in/BookAppointmentUseCase.java
package com.example.market.application.port.in;

import com.example.market.application.command.BookAppointmentCommand;
import com.example.market.domain.inspection.scheduling.InspectionAppointment;

public interface BookAppointmentUseCase {
    InspectionAppointment book(BookAppointmentCommand command);
}

File: market-application/src/main/kotlin/com/example/market/application/port/in/BookAppointmentUseCase.kt
package com.example.market.application.port.`in`

import com.example.market.application.command.BookAppointmentCommand
import com.example.market.domain.inspection.scheduling.InspectionAppointment

interface BookAppointmentUseCase {
    fun book(command: BookAppointmentCommand): InspectionAppointment
}
